tokens/usecase/token_registry_test.go

```diff
@@ -0,0 +1,61 @@
+package usecase
+
+import (
+	"testing"
+
+	"github.com/osmosis-labs/sqs/domain"
+)
+
+func TestFetchAndUpdateTokens(t *testing.T) {
```

Comment:
This is very related with this comment: https://github.com/osmosis-labs/sqs/pull/323/files#r1665915262

I was not able to find elegant apprach how to write tests against `ChainRegistryHTTPFetcher` to check its internal state `initialHash` value without making test to live in the same packgae as t `ChainRegistryHTTPFetcher` implementation.  One of the options could possibly be to add public method to return `initialHash` but that would leak implementation details, with larger data structures that would result in too verbose API.

I understand that this approach is changing course of current approach and I am more than happy to align to it if we can came up with some ideas.